addons/mrp/models/mrp_production.py

```diff
@@ -936,6 +936,12 @@ def create(self, vals_list):
                 and vals.get('date_finished')
                 and rec.move_finished_ids[0].date != vals['date_finished']):
                 rec.move_finished_ids.write({'date': vals['date_finished']})
+            elif (rec.move_finished_ids
+                  and rec.date_finished
+                  and rec.move_finished_ids[0].date != rec.date_finished
+                  and not vals.get('date_finished')):
+                # if no value is specified, do take the workorder duration (etc) into account
+                rec.move_finished_ids.write({'date': rec.date_finished})
```

Comment:
Since this is another bug as indicated in [1], I'd suggest doing a separate commit for that specific fix and add a test for that case (or update an existing test). This is for readability when someone in the future will check why it was added.